### DISCUSSION TITLE
Add Xray sidecar container to an ECS Pattern (Fargate)

### DISCUSSION THREAD

**Discussion by Erry91** [aws/aws-cdk#27749] [Q&A]
Hi,

I was wondering if someone could provide an example on how to enable Xray tracing in my Fargate service. As of now the way it is being deployed is via ECS Patterns, in particular the [NetworkLoadBalancedFargateService](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ecs_patterns.NetworkLoadBalancedFargateService.html).

I followed these issues where the topic was mentioned  ( #14284 and #14625 ) but it wasn't clear if a solution was ever officially provided. 

Is it possible to do this at the moment with the current constructs in CDK? And, if so, is it possible to have a snippet of how to do it?

Thank you very much in advance.

**Comment by TrevorBurnham** [comment]
I searched around for an example of setting up an X-Ray tracing "sidecar container" for Fargate and eventually found one here: https://community.aws/content/2nNrv2XxSZHK8EUHXxCA3CToXTx/instrumenting-applications-on-aws-ecs-and-aws-fargate-with-aws-x-ray?lang=en

```ts
taskDefinition.addContainer("ProductsServiceContainer", {
   image: ecs.ContainerImage.fromEcrRepository(props.repository, "1.0.0"),
   containerName: "ProductsService",
   logging: logDriver,         
   portMappings: [{
      containerPort: 8080,
      protocol: ecs.Protocol.TCP          
   }],         
   cpu: 384,
   memoryLimitMiB: 896,
   environment: {
      PRODUCTS_DDB: productsDdb.tableName,
      AWS_XRAY_DAEMON_ADDRESS: "0.0.0.0:2000",
      AWS_XRAY_CONTEXT_MISSING: "IGNORE_ERROR",
      AWS_XRAY_TRACING_NAME: "products-service"           
   }      
})
```
